frontend/src/views/camp/Story.vue

```diff
@@ -87,12 +97,29 @@ export default {
       return `${PRINT_SERVER}/?camp=${this.camp().id}&pagedjs=true&${configGetParams}`
     }
   },
+  watch: {
+    period () {
+      this.setInitialExpandedDays()
+    }
+  },
   mounted () {
-    this.camp().periods()._meta.load.then(periods => {
-      this.openPeriods = periods.items
-        .map((period, idx) => Date.parse(period.end) >= new Date() ? idx : null)
-        .filter(idx => idx !== null)
-    })
+    this.setInitialExpandedDays()
+  },
+  methods: {
+    expandAll () {
+      this.expandedDays = [...Array(this.period().days().items.length).keys()]
+    },
+    collapseAll () {
+      this.expandedDays = []
+    },
+    setInitialExpandedDays () {
+      this.period().days()._meta.load.then(days => {
```

Comment:
This might be slightly easier to read when rewriting it with `await`:
```js
async setInitialExpandedDays () {
  const days = await this.period().days()._meta.load
  this.expandedDays = days.items.map(...)
```

You can still safely call the async method from `mounted` without any changes.